icepyx/tests/test_behind_NSIDC_API_login.py

```diff
@@ -49,8 +54,33 @@ def test_download_granules_with_subsetting(reg, session):
     reg.download_granules(path)
 
 
-# def test_download_granules_without_subsetting(reg_a, session):
-#     path = './downloads'
-#     reg_a.order_granules(session, subset=False)
-#     reg_a.download_granules(session, path)
-#     #check that the max extent of the downloaded granules isn't subsetted
+def test_download_granules_without_subsetting(reg, session, capsys):
+    """
+    Test that granules can be ordered from NSIDC and downloaded with the `subset=False`
+    option.
+    """
+    path = "./downloads"
+
+    reg.order_granules(verbose=False, subset=False, email=False)
+    out, err = capsys.readouterr()  # capture stdout and stderr
+    assert out.startswith(
+        "Total number of data order requests is  1  for  3  granules.\n"
+        "Data request  1  of  1  is submitting to NSIDC\n"
+    )
+    assert err == ""
+
+    assert reg.reqparams == {
+        "client_string": "icepyx",
+        "include_meta": "Y",
+        "page_num": 0,
+        "page_size": 2000,
+        "request_mode": "async",
+        "short_name": "ATL06",
+        "version": "006",
+    }
+    assert len(reg.granules.orderIDs) == 2
+    assert int(reg.granules.orderIDs[0]) >= 5_000_000_000_000
+
+    reg.download_granules(path=path)
+    assert len(glob.glob(pathname=f"{path}/processed_ATL06_201902*.h5")) == 3
+    # check that the max extent of the downloaded granules isn't subsetted
```

Comment:
> assert len(glob.glob(pathname=f"{path}/processed_ATL06_201902*.h5")) == 3

If the granules are NOT subset, would they still have `processed_` preceding the filename? I thought that wasn't the case, but would need to verify.

>   # check that the max extent of the downloaded granules isn't subsetted

Is checking the (approximate) file size a viable way to do this? I have a vague notion that I started writing this test because I noticed that all files requested via icepyx were getting subset, even when `subset=False` was set by the user. I think we'd need to check the non-setting of `reg.subsetparams` to ensure none of those are being passed, but I can't recall offhand if it's the use of CMR kwargs (vs subset kwargs) for space and time or something else that triggers in the API whether or not subsetting happens.

Reply:
Yeah, I thought they shouldn't have a `processed_` prefix too, wondering if it's because it's cached the previous (non-subsetted) order and so this is what is returned (seems like a bug)? And yes, good idea on checking the filesize, I didn't want to open the file to check the extent of the point cloud, and totally forgot about just checking the filesize :laughing: